qa/L0_data_compression/test.sh

```diff
@@ -70,7 +70,37 @@ fi
 set -e
 
 # End-to-end testing with simple model
+function run_data_compression_infer_client() {
+    local client_path=$1
+    local request_algorithm=$2
+    local response_algorithm=$3
+    local log_path=$4
+
+    local python_or_cpp=`echo -n "$client_path" | tail -c 3`
+    if [ "$python_or_cpp" == ".py" ]; then
+        local infer_client="python $client_path"
+        local request_algorithm_parameter="--request-compression-algorithm $request_algorithm"
+        local response_algorithm_parameter="--response-compression-algorithm $response_algorithm"
+    else  # C++
+        local infer_client=$client_path
+        local request_algorithm_parameter="-i $request_algorithm"
+        local response_algorithm_parameter="-o $response_algorithm"
+    fi
+
+    if [ "$request_algorithm" != "" ] && [ "$response_algorithm" == "" ]; then
+        $infer_client -v $request_algorithm_parameter >> $log_path 2>&1
+        return $?
+    fi
+    if [ "$request_algorithm" == "" ] && [ "$response_algorithm" != "" ]; then
+        $infer_client -v $response_algorithm_parameter >> $log_path 2>&1
+        return $?
+    fi
+    $infer_client -v $request_algorithm_parameter $response_algorithm_parameter >> $log_path 2>&1
```

Comment:
Also optional, but an alternative I find slightly quicker to parse is to build up the args and call the command in only one place, something like this:
```suggestion
    compression_args=""
    if [ "$request_algorithm" != "" ]; then
        compression_args+="${request_algorithm_parameter} "
    fi
    
    if [ "$response_algorithm" != "" ]; then
        compression_args+="${response_algorithm_parameter} "
    fi
    
    $infer_client ${compression_args} >> $log_path 2>&1
```

Reply:
Sure, build up the arguments in one place: [Improve command line option handling](https://github.com/triton-inference-server/server/pull/6975/commits/1d869967bcb7e540ef8c60eb845ef57a07335523)